File: su/src/test/java/og/sqlbuilder/sumo/TestProcessor.java
package og.sqlbuilder.sumo;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.sqlbuilder.su.KBLoader;
import org.sqlbuilder.su.SuProcessor;
import org.sqlbuilder.su.objects.Formula;
import org.sqlbuilder.su.objects.SUFile;
import org.sqlbuilder.su.objects.Term;

import static org.junit.jupiter.api.Assertions.fail;

@ExtendWith({KbLoaderExtension.class})
public class TestProcessor
{
	@Test
	public void testProcessFiles()
	{
		try // (SetCollector<SUMOFile> ignored = SUMOFile.COLLECTOR.open())
		{
			SuProcessor.insertFiles(TestUtils.OUT, SUFile.COLLECTOR, "files", "fileid,file,version,date");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessTermsAndAttrs()
	{
		try // (SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open())
		{
			SuProcessor.insertTermsAndAttrs(TestUtils.OUT, TestUtils.ERR, Term.COLLECTOR, KBLoader.kb, "terms", "sumoid,term", "terms_attr", "sumoid,attr");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessTermAttrs()
	{
		try // (SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open())
		{
			SuProcessor.insertTermAttrs(TestUtils.OUT, Term.COLLECTOR, KBLoader.kb, "terms_attr", "sumoid,attr");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessTerms()
	{
		try // (SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open())
		{
			SuProcessor.insertTerms(TestUtils.OUT, Term.COLLECTOR, "terms", "sumoid,term");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessFormulasAndArgs()
	{
		try //(
		//SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open(); //
		//SetCollector<SUMOFile> ignored2 = SUMOFile.COLLECTOR.open(); //
		//SetCollector<SUMOFormula> ignored3 = SUMOFormula.COLLECTOR.open(); //
		//)
		{
			SuProcessor.insertFormulasAndArgs(TestUtils.OUT, TestUtils.ERR, Formula.COLLECTOR, "formulas", "formulaid,formula,fileid", "formulas_args", "formulaid,sumoid,argtype,argnum");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessFormulas()
	{
		try //(
		//SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open(); //
		//SetCollector<SUMOFile> ignored2 = SUMOFile.COLLECTOR.open(); //
		//SetCollector<SUMOFormula> ignored3 = SUMOFormula.COLLECTOR.open(); //
		//)
		{
			SuProcessor.insertFormulas(TestUtils.OUT, Formula.COLLECTOR, "formulas", "formulaid,formula,fileid");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@Test
	public void testProcessFormulasArgs()
	{
		try //(
		//SetCollector<SUMOTerm> ignored = SUMOTerm.COLLECTOR.open(); //
		//SetCollector<SUMOFile> ignored2 = SUMOFile.COLLECTOR.open(); //
		//SetCollector<SUMOFormula> ignored3 = SUMOFormula.COLLECTOR.open(); //
		//)
		{
			SuProcessor.insertFormulaArgs(TestUtils.OUT, Formula.COLLECTOR, "formulas_args", "formulaid,sumoid,argtype,argnum");
		}
		catch (Exception e)
		{
			fail(e.getMessage());
		}
	}

	@BeforeAll
	public static void init()
	{
		SuProcessor.collectFiles(KBLoader.kb);
		SuProcessor.collectTerms(KBLoader.kb);
		SuProcessor.collectFormulas(KBLoader.kb);

		SUFile.COLLECTOR.open();
		Term.COLLECTOR.open();
		Formula.COLLECTOR.open();
	}

	@AfterAll
	public static void shutdown()
	{
		SUFile.COLLECTOR.close();
		Term.COLLECTOR.close();
		Formula.COLLECTOR.close();
	}

	public static void main(String[] args)
	{
		new KBLoader().load();
		init();
		TestProcessor p = new TestProcessor();
		p.testProcessFiles();
		p.testProcessTermsAndAttrs();
		p.testProcessFormulasAndArgs();
		shutdown();
	}
}

File: su/src/test/java/org/semantikos/sumo/TestProcessor.kt
package og.semantikos.sumo

import org.junit.jupiter.api.AfterAll
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.semantikos.su.KBLoader
import org.semantikos.su.SuProcessor
import org.semantikos.su.objects.Formula
import org.semantikos.su.objects.SUFile
import org.semantikos.su.objects.Term

@ExtendWith(KbLoaderExtension::class)
class TestProcessor {

    @Test
    fun testProcessFiles() {
        try {
            //SUFile.COLLECTOR.open().use {
                SuProcessor.insertFiles(TestUtils.OUT, SUFile.COLLECTOR, "files", "fileid,file,version,date")
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessTermsAndAttrs() {
        try {
            //Term.COLLECTOR.open().use {
                SuProcessor.insertTermsAndAttrs(TestUtils.OUT, TestUtils.ERR, Term.COLLECTOR, KBLoader.kb!!, "terms", "sumoid,term", "terms_attr", "sumoid,attr")
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessTermAttrs() {
        try {
            //Term.COLLECTOR.open().use {
                SuProcessor.insertTermAttrs(TestUtils.OUT, Term.COLLECTOR, KBLoader.kb!!, "terms_attr", "sumoid,attr")
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessTerms() {
        try {
            //Term.COLLECTOR.open().use {
                SuProcessor.insertTerms(TestUtils.OUT, Term.COLLECTOR, "terms", "sumoid,term")
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessFormulasAndArgs() {
        try {
            //Term.COLLECTOR.open().use {
            //    SUFile.COLLECTOR.open().use {
            //        Formula.COLLECTOR.open().use {
                        SuProcessor.insertFormulasAndArgs(TestUtils.OUT, TestUtils.ERR, Formula.COLLECTOR, "formulas", "formulaid,formula,fileid", "formulas_args", "formulaid,sumoid,argtype,argnum")
            //        }
            //    }
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessFormulas() {
        try {
            //Term.COLLECTOR.open().use {
            //    SUFile.COLLECTOR.open().use {
            //        Formula.COLLECTOR.open().use {
                        SuProcessor.insertFormulas(TestUtils.OUT, Formula.COLLECTOR, "formulas", "formulaid,formula,fileid")
            //        }
            //    }
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    @Test
    fun testProcessFormulasArgs() {
        try {
            //Term.COLLECTOR.open().use {
            //    SUFile.COLLECTOR.open().use {
            //        Formula.COLLECTOR.open().use {
                        SuProcessor.insertFormulaArgs(TestUtils.OUT, Formula.COLLECTOR, "formulas_args", "formulaid,sumoid,argtype,argnum")
            //        }
            //    }
            //}
        } catch (e: Exception) {
            Assertions.fail<Any?>(e.message)
        }
    }

    companion object {

        @JvmStatic
        @BeforeAll
        fun init() {
            SuProcessor.collectFiles(KBLoader.kb!!)
            SuProcessor.collectTerms(KBLoader.kb!!)
            SuProcessor.collectFormulas(KBLoader.kb!!)

            SUFile.COLLECTOR.open()
            Term.COLLECTOR.open()
            Formula.COLLECTOR.open()
        }

        @JvmStatic
        @AfterAll
        fun shutdown() {
            SUFile.COLLECTOR.close()
            Term.COLLECTOR.close()
            Formula.COLLECTOR.close()
        }

        @JvmStatic
        fun main(args: Array<String>) {
            KBLoader().load()
            init()
            val p = TestProcessor()
            p.testProcessFiles()
            p.testProcessTermsAndAttrs()
            p.testProcessFormulasAndArgs()
            shutdown()
        }
    }
}
